### ISSUE TITLE
vivado crashes when i launch synthesis, below is the terminal log

### ISSUE THREAD

**Issue by shahbaazlokh** [ichi4096/vivado-on-silicon-mac#1]
#
# An unexpected error has occurred (6)
#
Stack:
/lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7ffffde1a520]
/lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x12c) [0x7ffffde6ea7c]
/lib/x86_64-linux-gnu/libc.so.6(raise+0x16) [0x7ffffde1a476]
/lib/x86_64-linux-gnu/libc.so.6(abort+0xd3) [0x7ffffde007f3]
/lib/x86_64-linux-gnu/libc.so.6(+0x896f6) [0x7ffffde616f6]
/lib/x86_64-linux-gnu/libc.so.6(+0xa0d7c) [0x7ffffde78d7c]
/lib/x86_64-linux-gnu/libc.so.6(realloc+0x36c) [0x7ffffde7db2c]
/lib/x86_64-linux-gnu/libudev.so.1(+0x15707) [0x7fffede6f707]
/lib/x86_64-linux-gnu/libudev.so.1(+0x1bb1b) [0x7fffede75b1b]
/lib/x86_64-linux-gnu/libudev.so.1(+0x75ff) [0x7fffede615ff]
/lib/x86_64-linux-gnu/libudev.so.1(+0x7b6b) [0x7fffede61b6b]
/lib/x86_64-linux-gnu/libudev.so.1(+0x10192) [0x7fffede6a192]
/lib/x86_64-linux-gnu/libudev.so.1(+0x105d3) [0x7fffede6a5d3]
/lib/x86_64-linux-gnu/libudev.so.1(udev_enumerate_scan_devices+0x2a1) [0x7fffede6b341]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libXil_lmgr11.so(+0x10f927) [0x7ffff4f0f927]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libXil_lmgr11.so(xilinxd_52bd858d5acf2fc4+0x9) [0x7ffff4f0fd89]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libXil_lmgr11.so(+0xc6566) [0x7ffff4ec6566]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libXil_lmgr11.so(xilinxd_52bd853912de43c2+0xc8) [0x7ffff4ec6098]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libXil_lmgr11.so(+0xb33a2) [0x7ffff4eb33a2]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libXil_lmgr11.so(xilinxd_52bd995765656b48+0x2a) [0x7ffff4ebd5da]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libXil_lmgr11.so(xilinxd_52bd700d1bd3c616+0x73) [0x7ffff4ebd6c3]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commonxillic.so(XilReg::Utils::GetHostInfo[abi:cxx11](XilReg::Utils::HostInfoType, bool) const+0x1a0) [0x7ffff971fda0]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commonxillic.so(XilReg::Utils::GetHostInfoFormatted[abi:cxx11](XilReg::Utils::HostInfoType, bool) const+0x59) [0x7ffff9726379]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commonxillic.so(XilReg::Utils::GetHostInfo[abi:cxx11]() const+0x103) [0x7ffff9726633]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commonxillic.so(XilReg::Utils::GetRegInfo(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool)+0x96) [0x7ffff972a0a6]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commonxillic.so(XilReg::Utils::GetRegInfoWebTalk(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x60) [0x7ffff972a2d0]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_project.so(HAPRWebtalkHelper::getRegistrationId[abi:cxx11]() const+0x3d) [0x7fffda060acd]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_project.so(HAPRWebtalkHelper::HAPRWebtalkHelper(HAPRProject*, HAPRDesign*, HWEWebtalkMgr*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x178) [0x7fffda0631c8]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_tcltasks.so(+0x1d611c5) [0x7fffefd891c5]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_tcltasks.so(+0x1d6a244) [0x7fffefd92244]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_common.so(+0xb9430f) [0x7fffff3ec30f]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(+0x3356f) [0x7ffff8c3356f]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(+0x76945) [0x7ffff8c76945]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(+0x7e0f9) [0x7ffff8c7e0f9]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(TclEvalObjEx+0x76) [0x7ffff8c35216]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_common.so(+0xb91c33) [0x7fffff3e9c33]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(Tcl_ServiceEvent+0x7f) [0x7ffff8ca7bef]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(Tcl_DoOneEvent+0x154) [0x7ffff8ca7f24]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commontasks.so(+0x2b3c27) [0x7ffff2311c27]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commontasks.so(+0x2bc4f7) [0x7ffff231a4f7]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commontasks.so(+0x2bcdff) [0x7ffff231adff]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_common.so(+0xb9430f) [0x7fffff3ec30f]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(+0x3356f) [0x7ffff8c3356f]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(+0x76945) [0x7ffff8c76945]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(+0x7e0f9) [0x7ffff8c7e0f9]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(TclEvalObjEx+0x76) [0x7ffff8c35216]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_commonmain.so(+0xc3d8) [0x7ffffffb43d8]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/libtcl8.5.so(Tcl_Main+0x1d0) [0x7ffff8ca02f0]
/home/user/Xilinx/Vivado/2022.2/lib/lnx64.o/librdi_common.so(+0xbc17ab) [0x7fffff4197ab]
/lib/x86_64-linux-gnu/libc.so.6(+0x94b43) [0x7ffffde6cb43]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x44) [0x7ffffdefdbb4]

**Comment by ichi4096** [comment]
I've encountered this behaviour before and it seems to have something to do with shared libraries and udev as seen [here](https://support.xilinx.com/s/question/0D54U00005Sgst2SAB/failed-batch-mode-execution-in-linux-docker-running-under-windows-host?language=en_US) and [here](https://community.flexera.com/t5/InstallAnywhere-Forum/Issues-when-running-Xilinx-tools-or-Other-vendor-tools-in-docker/m-p/245820#M10647). However, I've set the LD_PRELOAD variable as an environment variable in the Dockerfile, which fixed it on my machine. I've added into the start_vivado.sh script that it should load the settings64.sh script; try again and let me know if this helps (you can simply replace start_vivado.sh with the new version on Github, no need to reinstall Vivado again). If this issue persists, could you send me an example of a design that makes Vivado crash?

**Comment by shahbaazlokh** [comment]
hi,

thanks for quick reply,

i tried to update start_vivado.sh script and its same problem.
i have attached design_1 tcl file, before you run this file make sure to rename this file as design_1.tcl and source this file after you launch vivado.

[design_1.txt](https://github.com/ichi4096/vivado-on-silicon-mac/files/11243098/design_1.txt)

**Comment by ichi4096** [comment]
When I source that file i get

ERROR: [Board 49-71] The board_part definition was not found for xilinx.com:kr260_som_som240_2_connector_kr260_carrier_som240_2_connector_som240_1_connector_kr260_carrier_som240_1_connector:part0:1.1. The project's board_part property was not set, but the project's part property was set to xck26-sfvc784-2LV-c. Valid board_part values can be retrieved with the 'get_board_parts' Tcl command. Check if board.repoPaths parameter is set and the board_part is installed from the tcl app store.

in the Tcl console. Do I need to install something additional for this to work?

**Comment by shahbaazlokh** [comment]
yes, there is a problem with this board selection, i have seen this on my linux machine as well,
no i have not installed anything manually.

however, i made a seprate example design with ac701 board with tcl script, attached below.

try this and let me know if the script can run and make block design.

[design_2.txt](https://github.com/ichi4096/vivado-on-silicon-mac/files/11244340/design_2.txt)

**Comment by ichi4096** [comment]
I was able to open this design and it also crashed with that error message, which is weird since the environment variables (specifically LD_PRELOAD) from the Dockerfile should prevent that. It turns out that some of those variables are not passed to the non-root user, which starts Vivado.

I've added the appropriate lines into the start_vivado.sh script, so try it with the new one again. Also, increase the Swap in Docker dashboard -> Settings -> Resources to 2 GB, since Docker assigns too little by default and Vivado will crash otherwise.

Additionally, execute the command `defaults write org.xquartz.X11 enable_iglx -bool true` in your terminal once. This should prevent any graphics glitches, which I discovered during synthesizing.
